test/e2e/deployment.go

```diff
@@ -486,8 +540,56 @@ func (d *Deployment) EnsureResourcesForLocalContour() error {
 		return err
 	}
 
+	if d.EnvoyDeploymentMode == DaemonsetMode {
```

Comment:
could this be a mutator func that takes a pointer to `d.EnvoyDaemonSet.Spec.Template` or `d.EnvoyDeployment.Spec.Template` and modifies it as needed? just so we don't have to duplicate this